spinalcordtoolbox/centerline/core.py

```diff
@@ -86,6 +88,7 @@ def get_centerline(im_seg, algo_fitting='polyfit', minmax=True, contrast=None, d
         # Compute derivatives using polynomial fit due to undefined derivatives using linear interpolation
         _, x_centerline_deriv = curve_fitting.polyfit_1d(z_mean, x_mean, z_ref, deg=degree)
         _, y_centerline_deriv = curve_fitting.polyfit_1d(z_mean, y_mean, z_ref, deg=degree)
+        z_centerline_deriv = np.ones_like(z_ref)
 
     elif algo_fitting == 'nurbs':
         from spinalcordtoolbox.centerline.nurbs import b_spline_nurbs
```

Comment:
@benjamindeleener I'm trying to wrap my head around the fact that `z_centerline_deriv` from nurbs is not a vector of ones, given that the space between each element of `z_centerline_fit` is one.

Reply:
ok i figured this out. This is because NURBS alters the density of sampling, and then outputs the derivatives (which are scaled by the density). In this commit (978a9c24b3878ec9a7de2a7dab2d793a2ae7616a) I have normalized the x and y derivatives by the z-derivatives, such that all algorithms in `get_centerline()` output comparable derivatives which can readily be used (e.g. for straightening, correcting for CSA, etc.).